src/dbnode/persist/fs/seek.go

```diff
@@ -372,6 +373,85 @@ func (s *seeker) SeekByIndexEntry(
 	return buffer, nil
 }
 
+func (s *seeker) SeekReadMismatchesByIndexChecksum(
+	checksum schema.IndexChecksum,
+	mismatchChecker wide.EntryChecksumMismatchChecker,
+	resources ReusableSeekerResources,
+) (wide.ReadMismatch, error) {
+	completed := false
+	defer func() {
+		// NB: if this fails to complete, finalize the checksum.
+		if !completed {
+			checksum.Finalize()
+		}
+	}()
+
+	// NB: first, apply the reader.
+	allMismatches, err := mismatchChecker.ComputeMismatchesForEntry(checksum)
+	if err != nil {
+		// NB: free checksum resources
+		return wide.ReadMismatch{}, err
+	}
+
+	// NB: only filter out reader side mismatches. TODO: remove index checksum
+	// mismatches, since they are not necessary in the updated model.
+	mismatches := allMismatches[:0]
+	for _, m := range allMismatches {
+		if !m.IsReaderMismatch() {
+			mismatches = append(mismatches, m)
+		}
+	}
+
+	if len(mismatches) == 0 {
+		// This entry matches; no need to retrieve data.
+		return wide.ReadMismatch{}, nil
+	}
+
+	if len(mismatches) > 1 {
+		return wide.ReadMismatch{}, fmt.Errorf("multiple reader mismatches")
```

Comment:
Maybe include the number for debugging if this happens?
```go
fmt.Errorf("expected only single reader mismatch: actual=%d", len(mismatches))
```

Reply:
This is one of the bits that'll get refactored (it'll return at most 1 entry soon), so may leave it for now